### ISSUE TITLE
Custom post type should get connected to users if it supports authors.

### ISSUE THREAD

**Issue by CodeProKid** [wp-graphql/wp-graphql#1041]
When registering a custom post type, we add connections to things like terms and featured media if they support those things, but we don't add a connection on users to that post type if the post type supports `authors`. Let's add that connection dynamically when the post type is registered.

**Comment by jasonbahl** [comment]
Still need to make this a one-to-one connection. 🤔

**Comment by jasonbahl** [comment]
@CodeProKid it looks like `author` is provided to _all_ Post Types as it stands. 

Are you saying you want `author` field to _not_ exist on Post Types that have not registered support for `authors`? compared to the current default of showing it for _all_ Post Types?

Related to: https://github.com/wp-graphql/wp-graphql/issues/405

**Comment by CodeProKid** [comment]
nope, I'm talking about the opposite side of the connection. So any post type supporting the `author` field should get connected on the `user` type, like posts do by default. Like:
```graphql
{
  users {
    edges {
      node {
        id
        customPostType {
          edges {
            node {
              id
              title
            }
          }
        }
      }
    }
  }
}
```

**Comment by jasonbahl** [comment]
Ahhh! I see.

**Comment by jasonbahl** [comment]
I think I read: 

> add a connection on users

as 

> add a connection _to_ users

**Comment by jasonbahl** [comment]
@CodeProKid this is addressed in #1337. 

Take this example:

```
add_action( 'init', function() {

	register_post_type( 'test', [
		'label' => 'Test',
		'show_ui' => true,
		'show_in_graphql' => true,
		'supports' => [ 'author' ],
		'graphql_single_name' => 'Test',
		'graphql_plural_name' => 'Tests'
	] );

} );
```

This will allow me to query like so: 

![Screen Shot 2020-06-04 at 12 55 39 PM](https://user-images.githubusercontent.com/1260765/83799231-bd3b3000-a662-11ea-889c-71dde9df525a.png)

**Query:**

```graphql
{
  viewer {
    id
    tests {
      nodes {
        id
      }
    }
  }
}
```

**Results:**

```json
{
  "data": {
    "viewer": {
      "id": "dXNlcjox",
      "tests": {
        "nodes": [
          {
            "id": "cG9zdDo2MjY="
          },
          {
            "id": "cG9zdDo1MzM="
          }
        ]
      }
    }
  }
}
```
